### PR TITLE
ViewModelProviders is deprecated

### PR DESCRIPTION
**Pull request by jtavva**
ViewModelProviders class is deprecated, code is updated to use ViewModelProvider constructors directly Fix for the issue #24

### CODE REVIEW

File: RecyclerViewClickHandler-Starter/build.gradle [hunk 1]
```diff
@@ -24,7 +24,7 @@ buildscript {
         room_version = '2.2.0'
         coroutine_version = '1.3.7'
         gradleVersion = '4.0.1'
-        navigationVersion = '1.0.0'
+        navigationVersion = '1.0.0-alpha07'
```
Comment (kkuan2011) [inline RecyclerViewClickHandler-Starter/build.gradle:27]: Is it 2.3.0 now? https://developer.android.com/jetpack/androidx/releases/navigation

File: RecyclerViewDiffUtilDataBinding-Starter/app/build.gradle [hunk 2]
```diff
@@ -62,7 +62,7 @@ dependencies {
     implementation "androidx.room:room-runtime:$room_version"
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
     kapt "androidx.room:room-compiler:$room_version"
-    implementation "androidx.lifecycle:lifecycle-extensions:2.2.0"
+    implementation "androidx.lifecycle:lifecycle-extensions:2.0.0"
```
Comment (kkuan2011) [inline RecyclerViewDiffUtilDataBinding-Starter/app/build.gradle:65]: how come it's down to 2.0.0?

File: RecyclerViewDiffUtilDataBinding-Starter/build.gradle [hunk 3]
```diff
@@ -24,7 +24,7 @@ buildscript {
         room_version = '2.2.0'
         coroutine_version = '1.3.7'
         gradleVersion = '4.0.1'
-        navigationVersion = '1.0.0'
+        navigationVersion = '1.0.0-alpha07'
```
Comment (kkuan2011) [inline RecyclerViewDiffUtilDataBinding-Starter/build.gradle:27]: Same question on nav version

File: RecyclerViewClickHandler-Starter/build.gradle (hunk 1 above)
Comment (jtavva) [reply RecyclerViewClickHandler-Starter/build.gradle:27]: This need some code changes and may be codelab changes as well. So it is best to do it in a separate CL

File: RecyclerViewDiffUtilDataBinding-Starter/app/build.gradle (hunk 2 above)
Comment (jtavva) [reply RecyclerViewDiffUtilDataBinding-Starter/app/build.gradle:65]: This need some code changes and may be codelab changes as well. So it is best to do it in a separate CL

File: RecyclerViewDiffUtilDataBinding-Starter/build.gradle (hunk 3 above)
Comment (jtavva) [reply RecyclerViewDiffUtilDataBinding-Starter/build.gradle:27]: This need some code changes and may be codelab changes as well. So it is best to do it in a separate CL.

**Pull request merged**